src/functions/mod.ts

```diff
@@ -66,3 +106,39 @@ export class SlackFunctionDefinition<
     };
   }
 }
+
+export class ConnectorDefinition<
+  InputParameters extends ParameterSetDefinition,
+  OutputParameters extends ParameterSetDefinition,
+  RequiredInput extends PossibleParameterKeys<InputParameters>,
+  RequiredOutput extends PossibleParameterKeys<OutputParameters>,
+> extends BaseFunctionDefinition<
+  InputParameters,
+  OutputParameters,
+  RequiredInput,
+  RequiredOutput
+> {
+  type: ManifestFunctionType = "API";
+  constructor(
+    public definition: FunctionDefinitionArgs<
+      InputParameters,
+      OutputParameters,
+      RequiredInput,
+      RequiredOutput
+    >,
+  ) {
+    super(definition);
+  }
+
+  export(): ManifestFunctionSchema {
```

Comment:
They are slightly different, the `ConnectorFunctionDefinition` `export` has an additional `type` field, adding this to the base class would change the output of `SlackFunctionDefinition.export()`

Reply:
Aren't we assigning `type: "app"` in `SlackFunctionDefinition`, meaning that we can return the `type` on those functions in the output manifest?

Reply:
Currently `type: "app"` is in `SlackFunctionDefinition` but is not part of the `returnType` of `export` adding `type: "app"` to the `returnType` of `export` may affect some users and mays also affect some tests related to the `generated builtin`

We did talk about remove the affected tests in the past 🧠 

I would recommend adding `type: "app"` to the `returnType` of `export` and removing the affected tests in a separate PR?

Reply:
But in this PR aren't you adding support for "app" [in the `returnType` of `export`](https://github.com/slackapi/deno-slack-sdk/pull/176/files#diff-8d25a59e46a06143e923f890c73ca8de7fa24f95ed7bc8bac40219bc78ba17afR210)? We just never actually export it from Slack functions, only from Connector functions. I'm not sure what the purpose of assigning `type: "app"` in `SlackFunctionDefinition` is if we don't use it 🤔 I assumed that returning it in the output manifest would work because it is the default if no type is specified?

Reply:
Yes it does work, my plan was to add `type: "app"` to the export in a follow up PR, I tried to keep this PR as simple as possible and not change the SDK behavior at all

I would also want to create a `BuiltinFunctionDefinition` in this follow up PR and use it in the generation script 🏄 
Lmk what you think

Reply:
Curious what we'll get from `BuiltinFunctionDefinition` — if there's value it may make sense to keep that in the repo where built-ins exist long-term and not necessarily the SDK

Reply:
I see, this is a classic java pattern where the SDK defines an interface and the other packages implements it